unstructured_ingest/v2/processes/connectors/kafka.py

```diff
@@ -0,0 +1,306 @@
+import base64
+import json
+import socket
+from dataclasses import dataclass, field
+from pathlib import Path
+from time import time
+from typing import TYPE_CHECKING, Any, Dict, Generator, List, Optional
+
+from pydantic import Field, Secret, SecretStr
+
+from unstructured_ingest.error import (
+    DestinationConnectionError,
+    SourceConnectionError,
+    SourceConnectionNetworkError,
+)
+
+# TODO DEPRECATE IT ONCE FINISH THE DESTINATION
+from unstructured_ingest.interfaces import (
+    BaseDestinationConnector,
+    BaseIngestDoc,
+    IngestDocSessionHandleMixin,
+    WriteConfig,
+)
+from unstructured_ingest.utils.data_prep import batch_generator
+from unstructured_ingest.utils.dep_check import requires_dependencies
+from unstructured_ingest.v2.interfaces import (
+    AccessConfig,
+    ConnectionConfig,
+    Downloader,
+    DownloaderConfig,
+    FileData,
+    FileDataSourceMetadata,
+    Indexer,
+    IndexerConfig,
+    download_responses,
+)
+from unstructured_ingest.v2.logger import logger
+from unstructured_ingest.v2.processes.connector_registry import SourceRegistryEntry
+
+if TYPE_CHECKING:
+    from confluent_kafka import Consumer, Producer
+
+CONNECTOR_TYPE = "kafka"
+
+
+class KafkaAccessConfig(AccessConfig):
+    kafka_api_key: Optional[SecretStr] = Field(
+        description="Kafka API key to connect at the server", default=None
+    )
+    secret: Optional[SecretStr] = Field(description="", default=None)
+
+
+class KafkaConnectionConfig(ConnectionConfig):
+    access_config: Secret[KafkaAccessConfig]
```

Comment:
Let's not make the access config itself optional but instead default it to an instantiated `KafkaAccessConfig` model. Example: [pinecone](https://github.com/Unstructured-IO/unstructured-ingest/blob/main/unstructured_ingest/v2/processes/connectors/pinecone.py#L41)